src/components/views/dialogs/SetMxIdDialog.js

```diff
@@ -106,6 +106,15 @@ export default React.createClass({
     },
 
     _doUsernameCheck: function() {
+        // XXX: SPEC-1
+        // Check if username is valid
+        if (encodeURIComponent(this.state.username) !== this.state.username) {
```

Comment:
So is this needed because the username checking API sends the user name in the URI?